scripts/cc-cli/src/commands/registerAddress.ts

```diff
@@ -91,3 +91,94 @@ function errorMsg(s: string) {
 export function isValidPrivateKey(pk: string): boolean {
   return utils.isHexString(pk, 32);
 }
+
+export async function initCallerEthWallet(
+  options: OptionValues,
+): Promise<Wallet> {
+  try {
+    return await initWalletFromEnvOrPrompt("ETH_PRIVATE_KEY", options);
+  } catch (e) {
+    console.error(getErrorMessage(e));
+    process.exit(1);
+  }
+}
+
+async function initWalletFromEnvOrPrompt(
+  envVar: string,
+  options: OptionValues,
+): Promise<Wallet> {
+  const interactive = options.input;
+  const useMnemonic = options.ethMnemonic;
+  const inputName = useMnemonic ? "mnemonic" : "private key";
+  const generateWallet = useMnemonic
+    ? newWalletFromMnemonic
+    : newWalletFromPrivateKey;
+  const validateInput = useMnemonic ? isMnemonicValid : isValidPrivateKey;
+
+  if (!interactive && process.env[envVar] === undefined) {
+    throw new Error(
+      "Error: Must specify a private key using the environment variable ETH_PRIVATE_KEY or an interactive prompt",
+    );
+  }
+
+  if (process.env[envVar] !== undefined) {
+    const seed = process.env[envVar] as string;
+
+    if (!validateInput(seed)) {
+      throw new Error("Error: Private key is invalid");
```

Comment:
Error message is a bit misleading b/c the input can be in 2 formats. If you want to avoid having more conditional s you can change the message to something more generic like "Error: invalid input".